src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -0,0 +1,139 @@
+package fr.acinq.lightning.blockchain.electrum
+
+import fr.acinq.bitcoin.*
+import fr.acinq.lightning.utils.Connection
+import fr.acinq.lightning.utils.lightningLogger
+import fr.acinq.lightning.utils.sat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlin.native.concurrent.ThreadLocal
+
+data class WalletState(val pubKeyScripts: Map<ByteVector, Pair<List<UnspentItem>, PrivateKey?>>) {
+    val utxos: List<UnspentItem> = pubKeyScripts.flatMap { it.value.first }
+    val balance: Satoshi = utxos.sumOf { it.value }.sat
+    val spendable: List<Triple<ByteVector, PrivateKey, UnspentItem>> by lazy {
+        pubKeyScripts
+            .entries
+            .filter { it.value.second != null }
+            .flatMap {
+                it.value.first.map { utxo -> Triple(it.key, it.value.second!!, utxo) }
+            }
+    }
```

Comment:
@t-bast is this helper useful?

Reply:
Why do we need to have nullable private keys in the first place?
As a consumer of this, I don't want to be exposed to those, I'm only interested in utxos I can spend.
The ideal interface I'd like is:

1. A function that gives me a set of utxos I can spend (with an optional filter on a given address, if I only want the funds I swapped-in to a specific address)
2. A function that signs a transaction containing utxos I got from the first step (iterate through the transaction's inputs and sign all the inputs that belong to the wallet)

Does that work with the current model? In this model I won't need access to the private keys at all, so this should stay an internal implementation detail.

Reply:
> Why do we need to have nullable private keys in the first place?

Because it may also be used as a watch-only wallet, e.g. to monitor final addresses from channels.

> Does that work with the current model?

Not well, because the wallet is as generic as possible, in order to also cover the case above. That's also why I moved from tracking addresses to scriptpubkeys. Problem is, being to generic prevents me from doing methods like the one you are asking for. 

Maybe I chose a wrong approach and should instead just build a `p2wpkh` wallet.

Reply:
> Maybe I chose a wrong approach and should instead just build a p2wpkh wallet.

I think it would make sense to start with only support for `p2wpkh` indeed, we're always going to generate our receiving addresses so we fully choose the script subset we want, don't we? We can then later add support for `p2tr` without any script, and maybe at some point for other types of script, but it's really not clear whether we'll need it.

Reply:
Agreed, I'll try that.

Reply:
Added your requested 2. in 25a6c32, didn't do 1. but it's should be straightforward.

I feel like next step is to wait until we actually use it and improve then.